### DISCUSSION TITLE
How to pass a file `Buffer` to MQTT APIs

### DISCUSSION THREAD

**Discussion by hassan-xy** [zwave-js/zwave-js-ui#3691] [Q&A] [answered]
Hi guys,
I am tasked with performing OTA for zwave controller using our application(uses nodejs). I checked the API:
{"api":"firmwareUpdateOTW","args":[{"name":"zwave_ncp_serial_api_controller_BRD4207A.gbl","data":{}}]}

My question is How can I place zwave_ncp_serial_api_controller_BRD4207A.gbl file properly from my /root directory into the API?
I have tried /root/zwave_ncp_serial_api_controller_BRD4207A.gbl but I  get error message.

Please share any information that might be of relevance.
Thanks

**Comment by robertsLando** [comment]
@hassan-xy Why you don't use the UI instead of the API?

**Comment by hassan-xy** [comment]
Because i will place my device in a remote location and won’t have access to the UI. So I need to hit the API. Can you suggest something.

Thanks

**Comment by robertsLando** [comment, marked as answer]
The arg of `firmwareUpdateOTW` is `file` that is like:

```ts
interface FwFile {
	name: string
	data: Buffer
	target?: number
}
```

So you should provide that as argument. Remember that for buffers the format is `{"type": "Buffer", "data": [//the buffer data array] }`

So:

```json
{
    "api": "firmwareUpdateOTW",
    "args": [
        {
            "name": "zwave_ncp_serial_api_controller_BRD4207A.gbl",
            "data": {"type": "Buffer", "data": [//the buffer data array] }
        }
    ]
}
```
